src/core/unit-dependency-atom.c

```diff
@@ -82,6 +82,11 @@ static const UnitDependencyAtom atom_map[_UNIT_DEPENDENCY_MAX] = {
         [UNIT_PROPAGATES_STOP_TO]     = UNIT_ATOM_RETROACTIVE_STOP_ON_STOP |
                                         UNIT_ATOM_PROPAGATE_STOP,
 
+        [UNIT_ON_FAILURE_OF]          = UNIT_ATOM_PROPAGATE_EXIT_STATUS,
+
+        [UNIT_ON_SUCCESS_OF]          = UNIT_ATOM_PROPAGATE_EXIT_STATUS |
+                                        UNIT_ATOM_REFERENCES,
```

Comment:
hmm, why the UNIT_ATOM_REFERENCES?

Reply:
ah, read up the discussion from last time now, looks ok

Reply:
but i think UNIT_ON_FAILURE_OF should have it too then.

Reply:
Yep, it should. I was relying on the fact the OnFailureOf= service is not collected by default. But it's better to have symmetry to cover different CollectModes.